tests/models/test_baseoperatormeta.py

```diff
@@ -0,0 +1,133 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+from unittest.mock import patch
+
+import pytest
+
+from airflow.configuration import conf
+from airflow.models.baseoperator import BaseOperator, ExecutorSafeguard
+from airflow.operators.python import task, PythonOperator
+from airflow.utils.state import DagRunState
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+
+class HelloWorldOperator(BaseOperator):
+    def execute(self, context: Context) -> Any:
+        return f"Hello {self.owner}!"
+
+
+class TestExecutorSafeguard:
+    def setup_method(self):
+        ExecutorSafeguard.test_mode = False
+
+    def teardown_method(self, method):
+        ExecutorSafeguard.test_mode = conf.getboolean("core", "unit_test_mode")
+
+    @pytest.mark.db_test
+    def test_executor_when_classic_operator_called_from_dag(self, dag_maker):
+        with dag_maker() as dag:
+            HelloWorldOperator(task_id="hello_operator")
+
+        dag_run = dag.test()
+        assert dag_run.state == DagRunState.SUCCESS
+
+    @pytest.mark.db_test
+    def test_executor_when_classic_operator_called_from_decorated_task(self, dag_maker):
+        with dag_maker() as dag:
+
+            @task(task_id="task_id", dag=dag)
+            def say_hello(**context):
+                operator = HelloWorldOperator(task_id="hello_operator")
+                return operator.execute(context=context)
```

Comment:
Now I think a bit more about it, I think this counts as a breaking behviour change, so it needs to be behind a config option which only warns by default.

Reply:
That what I also proposed at first but @potiuk wanted it to make it difficult to bypass (otherwise they will ignore it by default anyway) and thus poposed a parameter in the operator to still make it possible to ignore it and just print a warning.  So I don't know what we have to do now

Reply:
This is normal to have differing opinions.

As usual -when we have difference in opinions we have to attempt to reach consensus. 

I am OK with warning (for sure for now) - but I would also like to explore the option to make it actually works (see the discussion).

Reply:
Yeah. let's have a `default_allow_nested_opertors` (if that name will be acceptable?) and `true` should be the default value. This way also MAYBE in the future we will add template `pre-processing` in this case and remove warning and turn it into `feature`